### PR TITLE
Better pooling interface

### PR DESCRIPTION
**Pull request by pimeys**
- Unify `Manager` impls into one `QuaintManager`
- Maybe have a better way of creating a pool with an url without exposing tokio-resource-pool items

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -248,8 +248,24 @@ impl Quaint {
...
+            inner: self.inner.check_out().await?,
+        })
+    }
+}
+
+/// A connection from the pool. Implements
+/// [Queryable](connector/trait.Queryable.html).
+pub struct PooledConnection {
+    inner: CheckOut<QuaintManager>,
+}
```
Comment (tomhoule) [inline src/lib.rs]: :+1:

File: .envrc [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export TEST_MYSQL=mysql://root:prisma@localhost:3306/prisma
```
Comment (pimeys) [inline .envrc]: @tomhoule

File: src/connector/mysql.rs [hunk 3]
```diff
@@ -339,29 +339,14 @@ impl Queryable for Mysql {
...
-    fn get_admin_config() -> OptsBuilder {
-        let mut config = OptsBuilder::new();
-        config.ip_or_hostname(env::var("TEST_MYSQL_HOST").unwrap());
-        config.tcp_port(env::var("TEST_MYSQL_PORT").unwrap().parse::<u16>().unwrap());
-        config.db_name(env::var("TEST_MYSQL_DB").ok());
-        config.pass(env::var("TEST_MYSQL_ROOT_PASSWORD").ok());
-        config.user("root".into());
-        config
+    lazy_static! {
+        static ref CONN_STR: String = env::var("TEST_MYSQL").unwrap();
```
Comment (pimeys) [inline src/connector/mysql.rs]: @tomhoule

**Comment by pimeys** [comment]
Fixes https://github.com/prisma/prisma2/issues/902 as an extra gift.

**Pull request merged**